components/automate-ui/src/app/pages/+compliance/compliance-reporting-sidebar/compliance-reporting-sidebar.component.spec.ts

```diff
@@ -1,35 +1,133 @@
+        MockComponent({ selector: 'app-landing',
+                        inputs: ['routePerms'] }),
+        MockComponent({ selector: 'app-authorized',
+                        inputs: ['allOf', 'anyOf'],
+                        template: '<ng-content></ng-content>' }),
+        MockComponent({ selector: 'chef-sidebar',
+                        template: '<ng-content></ng-content>' }),
+        MockComponent({ selector: 'chef-sidebar-entry',
+                        inputs: ['route', 'icon', 'exact'],
+                        template: '<ng-content></ng-content>' })
       ],
-      schemas: [
-        CUSTOM_ELEMENTS_SCHEMA
+      providers: [
+        FeatureFlagsService
       ]
-    })
-    .compileComponents();
-  }));
+    });
 
-  beforeEach(() => {
     fixture = TestBed.createComponent(ComplianceReportingSidebarComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    complianceLandingComponent =
+      TestBed.createComponent(ComplianceLandingComponent).componentInstance;
+    element = fixture.debugElement;
   });
 
-  it('exists', () => {
+  it('should be created', () => {
+    fixture.detectChanges();
     expect(component).toBeTruthy();
   });
+
+  describe('compliance sidebar', () => {
+
+    it('shows all links consistent with compliance-landing', () => {
+      fixture.detectChanges();
+      const links = element.nativeElement
+        .querySelectorAll('div.nav-items chef-sidebar-entry');
+      expect(links.length).toBe(complianceLandingComponent.routeList.length);
+    });
+
+    it('has route order consistent with compliance-landing', () => {
+      fixture.detectChanges();
+      const links = element.nativeElement
+        .querySelectorAll('div.nav-items chef-sidebar-entry');
+      for (let i = 0; i < complianceLandingComponent.routeList.length; i++) {
+        expect(links[i].getAttribute('route'))
+          .toBe(complianceLandingComponent.routeList[i].route);
+      }
+    });
+
+   it('has paths consistent with compliance-landing', () => {
+     fixture.detectChanges();
+     const elements = Array
+       .from<HTMLElement>(
+         element.nativeElement.querySelectorAll('div.nav-items app-authorized'))
+       .filter(elem =>
+         elem.firstElementChild && elem.firstElementChild.tagName === 'CHEF-SIDEBAR-ENTRY');
+
+     expect(elements.length).toBe(complianceLandingComponent.routeList.length);
+     for (let i = 0; i < complianceLandingComponent.routeList.length; i++) {
+       checkFirstPerm(
+         'anyOf',
+         elements[i].getAttribute('ng-reflect-any-of'),
+         complianceLandingComponent.routeList[i].anyOfCheck);
+       checkFirstPerm(
+         'allOf',
+         elements[i].getAttribute('ng-reflect-all-of'),
+         complianceLandingComponent.routeList[i].allOfCheck);
+     }
+   });
+  });
+
+  function checkFirstPerm(label: string, permsFromTemplate: string, permsFromLandingComp: Check[]) {
```

Comment:
Rather than duplicating this function (I suspect your changes should work fine for admin-sidebar, too), suggest move it into components/automate-ui/src/app/testing/spec-helpers.ts so it can be re-used.